File: sdks/ts/src/sdk/models/operations/postfoldersidmetadataenterprisesecurityclassification6vmvochwuwo.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PostFoldersIdMetadataEnterpriseSecurityClassification6VmVochwUWoPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=folder_id" })
  folderId: string;
}


export class PostFoldersIdMetadataEnterpriseSecurityClassification6VmVochwUWoRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=Box__Security__Classification__Key" })
  boxSecurityClassificationKey?: string;
}


export class PostFoldersIdMetadataEnterpriseSecurityClassification6VmVochwUWoRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PostFoldersIdMetadataEnterpriseSecurityClassification6VmVochwUWoPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PostFoldersIdMetadataEnterpriseSecurityClassification6VmVochwUWoRequestBody;
}


export class PostFoldersIdMetadataEnterpriseSecurityClassification6VmVochwUWoResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  classification?: shared.Classification;

  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: sdks/go/pkg/models/operations/postfoldersidmetadataenterprisesecurityclassification6vmvochwuwo.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PostFoldersIDMetadataEnterpriseSecurityClassification6VMVochwUWoPathParams struct {
	FolderID string `pathParam:"style=simple,explode=false,name=folder_id"`
}

type PostFoldersIDMetadataEnterpriseSecurityClassification6VMVochwUWoRequestBody struct {
	BoxSecurityClassificationKey *string `json:"Box__Security__Classification__Key,omitempty"`
}

type PostFoldersIDMetadataEnterpriseSecurityClassification6VMVochwUWoRequest struct {
	Retries    *utils.RetryConfig
	PathParams PostFoldersIDMetadataEnterpriseSecurityClassification6VMVochwUWoPathParams
	Request    *PostFoldersIDMetadataEnterpriseSecurityClassification6VMVochwUWoRequestBody `request:"mediaType=application/json"`
}

type PostFoldersIDMetadataEnterpriseSecurityClassification6VMVochwUWoResponse struct {
	Classification *shared.Classification
	ClientError    *shared.ClientError
	ContentType    string
	StatusCode     int64
}
